tests/unit/workflows/test_workflows_tasks_arxiv.py

```diff
@@ -181,6 +181,72 @@ def test_populate_arxiv_document_logs_on_pdf_not_existing():
         assert expected == result
 
 
+def test_populate_arxiv_document_documents_not_existing():
+    response500 = {'content': '', 'status_code': 500}
+    response200 = {
+        'content': pkg_resources.resource_string(
+            __name__, os.path.join('fixtures', '1707.02785.html')),
+        'status_code': 200,
+    }
+    with requests_mock.Mocker() as requests_mocker:
+        requests_mocker.get(
+            'http://export.arxiv.org/pdf/1707.02785',
+            (response200,),
+        )
+        requests_mocker.get(
+            'http://arxiv.org/pdf/1707.02785',
+            (response500,),
+        )
+        schema = load_schema('hep')
+        subschema = schema['properties']['arxiv_eprints']
+
+        data = {
+            'arxiv_eprints': [
+                {
+                    'categories': [
+                        'cs.CV',
+                    ],
+                    'value': '1707.02785',
+
+                },
+            ],
+            'documents': []
+        }  # literature/1458302
+        extra_data = {}
+        files = MockFiles({})
+        assert validate(data['arxiv_eprints'], subschema) is None
+
+        obj = MockObj(data, extra_data, files=files)
+        eng = MockEng()
+
+        assert populate_arxiv_document(obj, eng) is None
+        assert not obj.data.get('documents')
```

Comment:
just last thing - can you assert `'documents' not in obj.data` (and same for figures? Cause here we don't know if the key is not there or maybe if the list is empty and we actually don't want this key to be present in wf